lua/auto-session/config.lua

```diff
@@ -20,7 +20,9 @@ local M = {}
 ---@field git_auto_restore_on_branch_change? boolean Should we auto-restore the session when the git branch changes. Requires git_use_branch_name
 ---@field lazy_support? boolean Automatically detect if Lazy.nvim is being used and wait until Lazy is done to make sure session is restored correctly. Does nothing if Lazy isn't being used. Can be disabled if a problem is suspected or for debugging
 ---@field bypass_save_filetypes? table List of file types to bypass auto save when the only buffer open is one of the file types listed, useful to ignore dashboards
+---@field close_filetypes_on_save? table Buffers with matching filetypes will be closed before saving, set to { 'checkhealth' } by default
 ---@field close_unsupported_windows? boolean Whether to close windows that aren't backed by a real file
+---@field preserve_buffer_on_restore? should_preserve_buffer_fn Called for each buffer, if it returns true, that buffer is preserved when restoring a session
```

Comment:
I tend to prefer configs like this being capable of receiving either a `fn` _or_ a `boolean`, so a user can just blanket set `preserve_buffer_on_restore = true` as well as `preserve_buffer_on_restore = function() ... end`

Reply:
The challenge with this particular config option is that if we allowed setting it to `true`, that would preserve all buffers when restoring which seems pretty surprising/likely to not be what someone would want. Since this is a fairly niche feature, my thinking was that it would be less error prone to require the user make a callback function vs someone misunderstanding what the option does, setting it to true, and then being surprised/confused.